profiling/src/profile/mod.rs

```diff
@@ -1155,6 +1320,1224 @@ mod api_test {
         profile.add(sample).unwrap_err();
     }
 
+    #[test]
```

Comment:
yeah maybe 😅

Reply:
This is definitely a lot of code -- this file grows by more than 50% with this change. I've left a few suggestions below of repeated things but... I think the overall tests do lose a bit by having so much repetition (e.g. it's hard to see what changes from test to test).

Reply:
Sorry I tried to make it easy. I followed what was done for the other tests, naively thinking that others would understand (except me because not enough experience in Rust 😅 )

Reply:
Well I probably have less Rust-foo than you, so that's part of the trouble hahaha :)